src/CoreWCF.Http/src/CoreWCF/Channels/HttpRequestContext.cs

```diff
@@ -425,22 +434,29 @@ protected override Task OnCloseAsync(CancellationToken token)
                 //}
             }
 
-            private async Task<ServiceSecurityContext> CreateSecurityContextAsync(IIdentity identity)
+            private async Task<ServiceSecurityContext> CreateSecurityContextAsync(IPrincipal principal)
             {
-                if (identity is WindowsIdentity wid)
+                if (principal.Identity is WindowsIdentity wid)
                 {
                     WindowsSecurityTokenAuthenticator tokenAuthenticator = new WindowsSecurityTokenAuthenticator();
                     SecurityToken windowsToken = new WindowsSecurityToken(wid);
                     ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = await tokenAuthenticator.ValidateTokenAsync(windowsToken);
                     return new ServiceSecurityContext(authorizationPolicies);
                 }
-                else if (identity is GenericIdentity gid)
+                else if (principal.Identity is GenericIdentity gid)
                 {
                     WindowsSecurityTokenAuthenticator tokenAuthenticator = new WindowsSecurityTokenAuthenticator();
                     SecurityToken genericToken = new GenericIdentitySecurityToken(gid, SecurityUniqueId.Create().Value);
                     ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = await tokenAuthenticator.ValidateTokenAsync(genericToken);
                     return new ServiceSecurityContext(authorizationPolicies);
                 }
+                else if (principal.Identity is ClaimsIdentity)
+                {
+                    AuthorizationContext authorizationContext = AuthorizationContext.CreateDefaultAuthorizationContext(null);
+                    authorizationContext.Properties.Add("ClaimsPrincipal", principal);
```

Comment:
@mconnew Here the magic string `"ClaimsPrincipal"` refers to the constant `CoreWCF.IdentityModel.Tokens.AuthorizationPolicy.ClaimsPrincipalKey`.
Please do notice that this const is internal to CoreWCF.Primitives and is used to pull `ClaimsPrincipal` from `AuthorizationContext` in 2 scenarii:
- The scenario of this PR (using ASP.NET Core AuthZ)
- Another scenario that uses IdentityModel

Reply:
Although strictly speaking the principal might not be a ClaimsPrincipal, it likely is if the Identity is a ClaimsIdentity. What do you think of using `nameof(ClaimsPrincipal)` here?

Reply:
done here https://github.com/CoreWCF/CoreWCF/pull/850/commits/5ac13a79cedcd259ff806e710c6cd3e51640949c